### PR TITLE
feat(consensus): Transaction for OpTxEnvelope

### PR DESCRIPTION
**Pull request by refcell**
### Description

Implements the [`alloy_consensus::Transaction`](https://docs.rs/alloy-consensus/latest/alloy_consensus/transaction/trait.Transaction.html) trait for the `OpTxEnvelope` type.

This exposes a gas limit accessor that should be used by `kona-executor` instead of [its custom method](https://github.com/anton-rs/kona/blob/main/crates/executor/src/util.rs#L61-L71).

### CODE REVIEW

**Review by clabby — APPROVED** [review]

**Comment by refcell** [comment]
Will wait for mattsse to give approval :)

**Review by mattsse — CHANGES_REQUESTED** [review]
looks like we're actually missing this impl:

https://github.com/alloy-rs/alloy/blob/de2a13340303db6dde1296f68df57078a5b1fdd8/crates/consensus/src/transaction/envelope.rs#L369-L369

we should do that instead

**Comment by refcell** [comment]
> alloy-rs/alloy@`de2a133`/crates/consensus/src/transaction/envelope.rs#L369-L369

Oh nice! Will do.

**Review by mattsse — APPROVED** [review]

**Pull request merged**
